R/fimsframe.R

```diff
@@ -154,6 +172,60 @@ setMethod(
   }
 )
 
+#' Get the length-composition data to be used in the model
+#'
+#' @param x The object containing the length-composition data.
+#' @param fleet_name The name of the fleet for the length-composition data.
+#' @export
+setGeneric("m_lengthcomp", function(x, fleet_name) standardGeneric("m_lengthcomp"))
+
+
+#' Get the length-composition data data to be used in the model
+#'
+#' @param x  The FIMSFrame containing length-composition data.
+#' @param fleet_name  The name of the fleet for the length-composition data.
+#' @export
+setMethod(
+  "m_lengthcomp", "FIMSFrame",
+  function(x, fleet_name) {
+    dplyr::filter(
+      .data = x@data,
+      .data[["type"]] == "length-comp",
+      .data[["name"]] == fleet_name
+    ) |>
+      dplyr::pull(as.numeric(.data[["value"]]))
+  }
+)
+
+#' Get the length at age conversion matrix to be used in the model
+#'
+#' @param x The object containing the length-composition data.
+#' @param fleet_name The name of the fleet for the length-composition data.
+#' @export
+setGeneric("m_length_at_age", function(x, fleet_name) standardGeneric("m_length_at_age"))
+
+#' Get the length at age conversion matrix to be used in the model
+#'
+#' @param x The object containing the length-composition data.
+#' @param fleet_name The name of the fleet for the length-composition data.
+#' @export
+setMethod(
+  "m_length_at_age", "FIMSFrame",
+  function(x, fleet_name) {
+    if("len" %in% colnames(x@data)){
+      dplyr::filter(
+        .data = as.data.frame(x@data),
+        .data[["type"]] == "length-at-age",
+        .data[["name"]] == fleet_name
+      ) |>
+        dplyr::group_by(.data[["age"]],.data[["len"]]) |>
+        dplyr::summarize(mean_value = mean(as.numeric(.data[["value"]]),
+        na.rm=TRUE)) |>
+        dplyr::pull(as.numeric(.data[["mean_value"]]))
+    }
```

Comment:
done, it's now m_age_to_length_conversion. added the "to" in the hopes that is a little clearer that it is ages converted to lengths (sums to 1 for each age) just in case we end up with a length-to-age-conversion as well at some point.